### PR TITLE
fix(ci): gh-runner.sh refactored to account for ARM architecture

### PR DESCRIPTION
**Pull request by adityasadalage**
## Problem
gh-runner.sh in contrib folder does not account for different installs in arm architecture

## Solution
Adding if condition checks to install the right .tar files in the existing runner script

### CODE REVIEW

**Review by skrdgraph — COMMENTED** [review]
This looks good ... did you test this out by attaching both kinds of runners?

File: contrib/gh-runner/gh-runner.sh [hunk 1]
```diff
@@ -22,9 +22,35 @@ sudo usermod -aG docker ${USER}
 newgrp docker &
 # Install & Setup GH Actions Runner
 mkdir actions-runner && cd actions-runner
-curl -o actions-runner-linux-x64-2.296.2.tar.gz -L https://github.com/actions/runner/releases/download/v2.296.2/actions-runner-linux-x64-2.296.2.tar.gz
-echo "34a8f34956cdacd2156d4c658cce8dd54c5aef316a16bbbc95eb3ca4fd76429a  actions-runner-linux-x64-2.296.2.tar.gz" | shasum -a 256 -c
-tar xzf ./actions-runner-linux-x64-2.296.2.tar.gz
+if [ "$(uname -m)" = "aarch64" ]; then
+    echo "Detected arm64 architecture"
+
```
Comment (skrdgraph) [inline contrib/gh-runner/gh-runner.sh:27]: pls remove empty lines*

Comment (adityasadalage) [reply contrib/gh-runner/gh-runner.sh]: Removed

File: contrib/gh-runner/gh-runner.sh [hunk 2]
```diff
@@ -22,9 +22,26 @@ sudo usermod -aG docker ${USER}
...
+elif [ "$(uname -m)" = "x86_64" ]; then
+    echo "Detected amd64 architecture"
+    # Download the latest runner package
+    curl -o actions-runner-linux-x64-2.298.2.tar.gz -L https://github.com/actions/runner/releases/download/v2.298.2/actions-runner-linux-x64-2.298.2.tar.gz
+    # Optional: Validate the hash
+    echo "0bfd792196ce0ec6f1c65d2a9ad00215b2926ef2c416b8d97615265194477117  actions-runner-linux-x64-2.298.2.tar.gz" | shasum -a 256 -c 
+    # Extract the installer
+    tar xzf ./actions-runner-linux-x64-2.298.2.tar.gz
+else
+    echo "Unrecognized architecture"
```
Comment (skrdgraph) [inline contrib/gh-runner/gh-runner.sh:42]: add `exit 1` here, I want the script to break when the architecture is unrecognized

Comment (adityasadalage) [reply contrib/gh-runner/gh-runner.sh:42]: Added change

**Review by skrdgraph — APPROVED** [review]

**Pull request merged**